enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/log/segmented/PositionCache.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.coreedge.raft.log.segmented;
+
+import org.neo4j.coreedge.raft.log.LogPosition;
+
+/**
+ * Caches (offsetIndex) -> (byteOffset) mappings, which can be used to find an exact or
+ * approximate byte position for an entry given an index. The index is defined as a relative
+ * offset index starting from 0 for each segment, instead of the absolute logIndex in the
+ * log file.
+ *
+ * The necessity and efficiency of this cache is understood by considering the values put into
+ * it. When closing cursors the position after the last entry is cached so that when the next
+ * batch of entries are to be read the position is already known.
+ */
+class PositionCache
+{
+    private static final LogPosition BEGINNING_OF_RECORDS = new LogPosition( 0, SegmentHeader.SIZE );
+    static final int CACHE_SIZE = 8;
+
+    private long[] index;
+    private long[] offset;
+    private int pos;
+
+    PositionCache()
+    {
+        index = new long[CACHE_SIZE];
+        offset = new long[CACHE_SIZE];
+    }
```

Comment:
We can remove the constructor and initialize fields in-line and make them `final`.